File: Data Structures & Algorithms/anagram-groups/submission-8.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> res = new HashMap<>();

        for(String word : strs){
           int[] count = new int[26];
           for(char c : word.toCharArray()){
            count[c - 'a']++;
           } 
           String strCount = Arrays.toString(count);
           System.out.println(strCount);

           res.putIfAbsent(strCount ,new ArrayList<>());
           res.get(strCount).add(word);
        }

        return new ArrayList<>(res.values());
    }
}

File: Data Structures & Algorithms/is-anagram/submission-8.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length) return false;

        var matchedRes = new List<char>();

        for(int c=0;c<s.Length; c++){
            for(int j = 0; j<t.Length; j++){
                if (s[c] == t[j]){
                    matchedRes.Add(s[c]);
                    t = t.Remove(j,1);
                    break;
                }
            }
        }

        return matchedRes.Count == s.Length;


    }
}
